my_methods.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+module Enumerable #:nodoc:
+  def my_each
+    # your code here
+    i = 0
+    while i < length
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield(i, self[i])
+      i += 1
+    end
+  end
+
+  def my_select
+    result = []
+    my_each { |element| result << element if yield(element) }
+  end
+
+  def my_all
+    flag = true
+    my_each { |element| flag &&= yield(element) }
+    flag
+  end
+
+  def my_any
+    flag = false
+    my_each { |element| flag ||= yield(element) }
+    flag
+  end
+
+  def my_none
+    flag = false
+    my_each { |element| flag ||= yield(element) }
+    !flag
+  end
+
+  def my_count
+    count = 0
+    my_each { |element| count += 1 if yield(element) }
+    count
+  end
```

Comment:
the functionality here is limited, it should work with an argument passed just like the main count method. Please check other functionalities